### PR TITLE
Make the vSphere work with distributed switches

### PR DESCRIPTION
**Pull request by mrIncompetent**
**What this PR does / why we need it**:
This makes the machine-controller work with distributed switches in vCenter.
Also it simplifies the logic to handle network adapters. 

```release-note
NONE
```

### CODE REVIEW

**Comment by mrIncompetent** [comment]
/retest

The ubuntu image was somehow configured with a static IP

File: pkg/cloudprovider/provider/vsphere/helper.go [hunk 1]
```diff
@@ -166,22 +180,6 @@ func createClonedVM(ctx context.Context, vmName string, config *Config, session
 		return nil, fmt.Errorf("failed to get virtual machine object after cloning: %v", err)
 	}
 
-	reconfigureTask, err := virtualMachine.Reconfigure(ctx, desiredConfig)
```
[the remark this answers is not in the document]
Comment (mrIncompetent) [reply pkg/cloudprovider/provider/vsphere/helper.go]: On what way does this aid debugging?

[the remark this answers is not in the document]
Comment (alvaroaleman) [reply pkg/cloudprovider/provider/vsphere/helper.go]: It tells you whether you don't have enough permission to clone or to reconfigure, the api just gives a "Permission denied" back which doesn't tell you permission _for what_ was denied

[the remark this answers is not in the document]
Comment (mrIncompetent) [reply pkg/cloudprovider/provider/vsphere/helper.go]: I'll update the code. Though it makes me cringe.

[the remark this answers is not in the document]
Comment (mrIncompetent) [reply pkg/cloudprovider/provider/vsphere/helper.go]: Done in https://github.com/kubermatic/machine-controller/pull/613/commits/f19eb729af6eef65130d4bf27b8860887bed5bed

[the remark this answers is not in the document]
Comment (alvaroaleman) [reply pkg/cloudprovider/provider/vsphere/helper.go]: I can understand that, initially it wasn't like that, then I had to debug vsphere permissions :S having ugly code is the smaller evil here IMHO

**Pull request merged**